docs/backup_and_restore.rst

```diff
@@ -123,67 +105,216 @@ Restoring
+   message ``ssh_exchange_identification: Connection closed by remote host``   
+   during the ``Wait for Tor to reload`` task. This is expected; during the 
+   restoration process, the Application Server's onion URL changed, causing it
+   to be unreachable.
+
+   Reboot the *Application Server*, or log in via the console and issue the
+   command ``sudo service tor reload`` to restart the Tor service.
+
+#. **Fix Tor credentials**: If you are using 
+   SSH-over-Tor and are re-using the *Admin Workstation* 
+   from your previous installation, you will need to remove the ``ssh`` 
+   alias created during re-installation, or you will see a host key 
+   verification failure warning in the next section.  
+
+   **Instances using v2 onion services:**
+
+   Inspect your ``app-ssh-aths`` file: 
+
+   .. code:: sh
+
+      cat install_files/ansible-base/app-ssh-aths
+
+   This file is in the format ``HidServAuth app_ssh_url Secret # client: admin``.
+   Type the following command, replacing ``app_ssh_url`` with the .onion URL 
+   found in the file above:
+
+   .. code:: sh
+
+      ssh-keygen -f "/home/amnesia/.ssh/known_hosts" -R "app_ssh_url"
+
+   **Instances using v3 onion services:**
+
+   Inspect your ``app-ssh.auth_private`` file: 
+
+   .. code:: sh
+
+      cat install_files/ansible-base/app-ssh.auth_private 
+
+
+   This file is in the format ``app_ssh_url:descriptor:x25519:secret``.
+   Type the following command, replacing ``app_ssh_url`` with the URL portion of
+   your file and appending ``.onion``, as shown below:
+
+   .. code:: sh
+
+      ssh-keygen -f "/home/amnesia/.ssh/known_hosts" -R "app_ssh_url.onion"
+
+   Then, edit the file called ``tor_v3_keys.json``, located in 
+   ``install_files/ansible-base`` to replace the two ``app_journalist`` and 
+   two ``app_ssh`` values in the file with their respective values 
+   that you preserved in the ``tor_v2_keys.json`` file from the previous installation.
+
+   .. important:: Do not change the ``mon_ssh`` values, since the restore process
+                  did not restore the Monitor server.  
+
+#. **Replace original Tor credentials**: Copy the following files preserved 
+   from the old installation into
+   ``install_files/ansible-base``, overwriting the existing files of the 
+   same name. (If you wish, you may temporarily save the existing files
+   to another location outside of the SecureDrop directory before overwriting 
+   them).  
+
+   **v2 onion services**
+
+   * ``app-source-ths``
+   * ``app-journalist-aths``
+   * ``app-ssh-aths``
+
+   **v3 onion services**
+
+   * ``app-sourcev3-ths``
+   * ``app-journalist.auth_private``
+   * ``app-ssh.auth_private``
+
+#. **Reconfigure Tor connections**: 
+   Run ``./securedrop-admin tailsconfig`` to update the *Admin Workstation*
+   to use the restored Onion URLs again. See :doc:`configure_admin_workstation_post_install`
+   for further instructions.
```

Comment:
Perhaps add a recommendation to verify connectivity here?

Reply:
Added in 151175d